examples/tests/valid/lift_inflight_closure_returning_object_issue6501.test.w

```diff
@@ -0,0 +1,17 @@
+// Some inflight object with a method
+inflight class Foo {
+  pub do():num {
+    return 1337;
+  }
+}
+
+let foo = inflight (): Foo => {
```

Comment:
Can you add a test similar to the bug report?

(Where the returned type is a preflight class)

Reply:
I added a commented out test because it fail (on sim, not on aws) due to #6500
@Chriscbr any idea?

Reply:
I pushed a change with a fix. It's sort of a hack to deal with the fact that we're using these proxy objects that implicitly forward any method call like `x.blah()` into an HTTP call. When you do `await x` in JavaScript, if you don't have a Promise object, then for backwards compatibility JavaScript will try calling a method named "then", hence the errors.

Reply:
Excellent, thanks!